source/ada/lsp-lal_utils.adb

```diff
@@ -57,14 +57,33 @@ package body LSP.Lal_Utils is
    -- Resolve_Name --
    ------------------
 
-   function Resolve_Name (Name_Node : Name) return Defining_Name is
-      --  In every case, if the unit has syntax errors, activate
-      --  Imprecise_Fallback.
-      --  TODO: Should we activate it in any case ?
-      Result : constant Defining_Name :=
-        Name_Node.P_Xref
-          (Imprecise_Fallback => Name_Node.Unit.Has_Diagnostics);
+   function Resolve_Name
+     (Name_Node : Name;
+      Imprecise : out Boolean) return Defining_Name
+   is
+      Result : Defining_Name;
    begin
+      --  Try to get the cross reference without enabling the imprecise
+      --  fallback first.
+      --  Property_Error exceptions can be raised so make sure to handle
+      --  them correctly.
+      declare
+      begin
+         Result := Name_Node.P_Xref (Imprecise_Fallback => True);
```

Comment:
shouldn't  we try first with  Imprecise_Fallback = False ?

Reply:
Wow, that was the purpose of my patch, thanks for noticing it